packages/happy-dom/src/nodes/element/Dataset.ts

```diff
@@ -46,10 +46,10 @@ export default class Dataset {
 				return true;
 			},
 			deleteProperty(dataset: DatasetRecord, key: string): boolean {
-				return (
-					!!element.attributes.removeNamedItem('data-' + Dataset.camelCaseToKebab(key)) &&
-					delete dataset[key]
-				);
+				if (element.attributes.removeNamedItem('data-' + Dataset.camelCaseToKebab(key))) {
```

Comment:
`removeNamedItem` might throw an [NotFoundError](https://developer.mozilla.org/en-US/docs/Web/API/NamedNodeMap/removeNamedItem#exceptions), maybe it should be wrapped with a `try catch`.

Reply:
@btea Just to be clear, you're talking about making this change for future compatibility with the spec, correct? Because currently, the Happy DOM implementation of `removeNamedItem` does not appear to ever throw that error. Please correct me if I'm wrong!

Reply:
Yes, as you said, the current implementation of Happy DOM will return `null` if the attribute does not exist.

So your current modification should be fine, maybe you can create a PR to modify the implementation compatibility specification of `removeNamedItem` in the future.

According to the standard, if the attribute does not exist, an error will be thrown directly. There should be no need to wrap it with `try catch` here. I thought wrong before.

Reply:
I would be happy to make that change in this PR if you'd like! I can make `removeNamedItem` throw an error if the item does not exist, and then make `deleteProperty` catch the error. Does that sound correct?

Reply:
I think this is correct, but `deleteProperty` should not need to catch errors, just throw them globally.

Reply:
My understanding is that deleting an item from a dataset should never throw an error, even if an item with that key does not exist.

```javascript
const el = document.createElement("div")
delete el.dataset.blah // returns true; no error thrown
```

So I think you were right originally and we should catch the error thrown by `removeNamedItem` in the `deleteProperty` method of the dataset proxy.

Reply:
Sorry, I confused `removeNamedItem` and `deleteProperty`, you are right. 👍

Reply:
I'll work on this tomorrow!

Reply:
@btea I just pushed those changes as discussed. Do you think this should still be considered a patch change? It technically might break someone's code if they were relying on `removeNamedItem` not ever throwing, but that behavior was also technically incorrect, so I'm not sure whether to consider it a breaking change. My own feeling is it should still be a patch, but I wanted to check with you.

Reply:
I think this should still be a patch change.